### DISCUSSION TITLE
Momentary scroll mode(Horizontal)

### DISCUSSION THREAD

**Discussion by tokuhirom** [Yowkees/keyball#561] [Ideas]
おしている間だけスクロール状態になると同時に scrolll snap mode が KEYBALL_SCROLLSNAP_MODE_HORIZONTAL になるキーが Special  key code にあると便利かなーと思いました。
scroll snap mode をいちいち horizontal にしてから vertical に戻す、という動作だとちょっと面倒なので。

雰囲気コードとしてはこんな感じで出来ると思います(自分が使ってる分にはこれでなかなか便利です)。
https://github.com/tokuhirom/keyball/commit/8ae41b881609df2692aa4daa899131a171976202

押下時に HORIZONTAL にして、離すときに VERTICAL にするという雑な挙動になってるので、このままだと公式のファームウェアには入れにくい仕様かなとは思いつつ。

なかなか便利かなーと思いましたので共有まで。
(自分は tweetdeck や excel などで横スクロールを多用するので、サッと使えると便利なのです)

**Comment by yoichi** [comment]
勝手に戻るようになってると、見えない状態（OLEDには見えてますが）に依存しなくてわかりやすいですね。
似たような考え方で、レイヤによってscroll snap modeを使い分けるようにしてみたら割と便利でした。自分はレイヤを一つ増やしているので3,4をそれぞれverticalとhorizontalに割り当ててみました。
https://github.com/yoichi/keyball/commit/f588d962fa71b86294e93d795317659a96206d23
